torch/distributed/fsdp/_init_utils.py

```diff
@@ -108,6 +108,8 @@ def _init_buffer_state(
     # since when buffer mixed precision is enabled, buffers are restored to
     # their original dtype for model checkpointing
     _buffer_name_to_orig_dtype: Dict[str, torch.dtype] = {}
```

Comment:
`_buffer_name_to_orig_dtype` maps each buffer name, prefixed starting from the root module passed to the FSDP constructor, to its original dtype. This is used for buffer mixed precision to restore the buffer to its original dtype during model checkpointing.

The previous implementation would save this information lazily upon the first call to `_cast_buffers()`. We can actually just pre-compute this information in the constructor.